deepchecks/utils/gower_distance.py

```diff
@@ -0,0 +1,96 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module for calculating distance matrix via Gower method."""
+
+import numpy as np
+import pandas as pd
+
+
+def gower_matrix(data: np.ndarray, cat_features: np.array) -> np.ndarray:
+    """
+    Calculate distance matrix for a dataset using Gower's method.
+
+    Can deal with missing values.
+    Requires each two samples to have at least one sheared non-null value.
+    Parameters
+    ----------
+    data: numpy.ndarray
+        Dataset matrix.
+    cat_features: numpy.array
+        Boolean array of representing which of the columns are categorical features.
+
+    Returns numpy.ndarray representing the distance matrix.
+    """
+    if not isinstance(data, np.ndarray): data = np.asarray(data)
+    ranges = calculate_ranges(data, cat_features)
+
+    result = np.zeros((data.shape[0], data.shape[0]))
+    for i in range(data.shape[0]):
```

Comment:
Or other forms of vectorization

Reply:
from vectorization: The [vectorize] function is provided primarily for convenience, not for performance. The implementation is essentially a for loop.
It will be less efficient in a vectorize manner due to double calculations for result[i, j], result[j, i]